nucypher/characters/base.py

```diff
@@ -84,6 +84,7 @@ def __init__(self,
                  signer: Signer = None,
                  registry: BaseContractRegistry = None,
                  include_self_in_the_state: bool = False,
+                 is_offchain: bool = False,
```

Comment:
Hmm - this kwarg becomes a bit superfluous IMO since it doesn't affect any other character's construction.  Alternately, I suggest relocating `provider_uri` into each `Character` subclass init.

Reply:
But because Bob is the only character that works without `provider_uri` we still want to raise `ValueError('Provider URI is required to init a decentralized character.')` in the character's respective init, correct?

Reply:
Yes, I that's why I'm thinking! 👍🏻